src/DOMNodeComparator.php

```diff
@@ -70,8 +70,15 @@ public function assertEquals($expected, $actual, $delta = 0.0, $canonicalize = f
     private function nodeToText(DOMNode $node, bool $canonicalize, bool $ignoreCase): string
     {
         if ($canonicalize) {
+            /** @psalm-var string|false $c14n */
+            $c14n = @$node->C14N();
+
+            if ($c14n === false) { // try node-to-text without canonicalize
+                return $this->nodeToText($node, !$canonicalize, $ignoreCase);
```

Comment:
Minor tweak: We should probably use a hardcoded `false` rather than performing a pointless inversion of a `bool` flag that one would have to scroll up for to find out whether or not it now will be `true` or `false`.